pkg/obfuscate/sql.go

```diff
@@ -299,10 +306,15 @@ func attemptObfuscation(tokenizer *SQLTokenizer) (*ObfuscatedQuery, error) {
 		out             = bytes.NewBuffer(make([]byte, 0, len(tokenizer.buf)))
 		err             error
 		lastToken       TokenKind
-		discard         = discardFilter{tokenizer.cfg.KeepSQLAlias}
-		replace         = replaceFilter{replaceDigits: tokenizer.cfg.ReplaceDigits}
-		grouping        groupingFilter
-		tableFinder     = tableFinderFilter{storeTableNames: storeTableNames}
+		metadata        SQLMetadata
+		discard         = discardFilter{
+			keepSQLAlias:    tokenizer.cfg.KeepSQLAlias,
+			collectComments: tokenizer.cfg.CollectComments,
+			metadata:        &metadata,
```

Comment:
Can we create a `metadataFinder` filter which extracts both the tables and the comments? Would that work? I think it makes more sense. This approach was quicker but a bit dirty IMHO.

Then, we could call something like `metadataFinder.Results()` which would return the `SQLMetadata`

Reply:
Thank you and great suggestion, this is indeed easier to read and maintain: https://github.com/DataDog/datadog-agent/pull/10042/commits/0522051bf95c902fb2a48a6099f4649d24985f27